### PR TITLE
POST & DELETE story by id

### PR DESCRIPTION
**Pull request by Armandosz117**
POST and DELETE story by id methods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
-# StoriesAPI
-StoriesAPI by Team Kingsman
+# StoriesAPI
+StoriesAPI by Team Kingsman
```
Comment (soydianapinto) [inline README.md:2]: Restore this file if you didn't change it at all.

File: src/main/java/com/stories/ApiApplication.java [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class ApiApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(ApiApplication.class, args);
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/ApiApplication.java:11]: Please remove the extra space and replace "ApiApplication" to "StoriesApplication".

File: src/main/java/com/stories/controller/AppController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.stories.domain.StoryDomain;
+import com.stories.service.StoryServiceImpl;
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/controller/AppController.java:15]: Please delete the extra space.

File: src/main/java/com/stories/controller/AppController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@RestController
+@RequestMapping(value = "/stories-qa.us-east-2.elasticbeanstalk.com/stories", produces = "application/json")
+public class AppController {
+	private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(AppController.class);
```
Comment (soydianapinto) [inline src/main/java/com/stories/controller/AppController.java:20]: Why are you adding this logger?

File: src/main/java/com/stories/controller/AppController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Autowired
+	StoryServiceImpl service;
```
Comment (soydianapinto) [inline src/main/java/com/stories/controller/AppController.java:23]: Please rename this instance to storyService

File: src/main/java/com/stories/controller/AppController.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	
+	@ResponseStatus(value = HttpStatus.CREATED)
+	@PostMapping("/createStory")
+	public void createStory(@RequestBody StoryDomain storyDomain) {
+		log.debug("Create user request - " + storyDomain.toString());
+		service.createStory(storyDomain);
+	}
+	
+	@ResponseStatus(value = HttpStatus.ACCEPTED)
```
Comment (soydianapinto) [inline src/main/java/com/stories/controller/AppController.java:32]: Please review the standard HTTP code response for a delete operation.

File: src/main/java/com/stories/controller/AppController.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	@DeleteMapping("/deleteStory")
+	public void deleteStory(@RequestParam(value="storyId", required=true) String storyId) {
+		service.deleteStory(storyId);
+    }
+	
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/controller/AppController.java:38]: Please delete the extra space.

File: src/main/java/com/stories/mapper/StoriesMapping.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+import ma.glasnost.orika.MapperFactory;
+import net.rakugakibox.spring.boot.orika.OrikaMapperFactoryConfigurer;
+
+public class StoriesMapping implements OrikaMapperFactoryConfigurer {
+	@Override
+	public void configure(MapperFactory orikaMapperFactory) {
+		orikaMapperFactory.classMap(StoryDomain.class, StoryDomain.class).byDefault().register();
+	}
+}
```
Comment (soydianapinto) [inline src/main/java/com/stories/mapper/StoriesMapping.java:13]: Where are you using this StoriesMapping class in your application?

File: src/main/java/com/stories/model/StoryModel.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Date;
+
+import org.bson.types.ObjectId;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+import lombok.Data;
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/model/StoryModel.java:13]: Please delete this extra space.

File: src/main/java/com/stories/model/StoryModel.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    private int progress;
+    private String status;
+    private String notes;
+    private String comments;
+    private Date start_date;
+    private Date due_date;
+    private String priority;
+    private String assignee_id;
+    private ArrayList history;
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/model/StoryModel.java:35]: Please delete this extra space.

File: src/main/java/com/stories/repository/StoriesRepository.java [hunk 11]
```diff
@@ -1,9 +1,17 @@
...
-import com.stories.model.Story;
+import com.stories.model.StoryModel;
+	
+public interface StoriesRepository extends MongoRepository<StoryModel, String> {
+	
+	List<StoryModel> findBytechnology(String technology);
+
+	void save(com.stories.domain.StoryDomain storyDomain);
+
+//	void save(com.stories.model.Story user);
```
Comment (soydianapinto) [inline src/main/java/com/stories/repository/StoriesRepository.java:15]: Please delete these if you are not using them.

File: src/main/java/com/stories/service/StoryService.java [hunk 12]
```diff
@@ -0,0 +1,10 @@
+package com.stories.service;
+
+import com.stories.domain.StoryDomain;
+
+//Especificacion del comportamiento
```
Comment (soydianapinto) [inline src/main/java/com/stories/service/StoryService.java:5]: Please delete this comment.

File: src/main/java/com/stories/service/StoryServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import com.stories.repository.StoriesRepository;
+
+import ma.glasnost.orika.MapperFacade;
+
+@Service
+public class StoryServiceImpl implements StoryService {
+	private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(StoryServiceImpl.class);
+	
+	@Autowired
+	StoriesRepository repository;
```
Comment (soydianapinto) [inline src/main/java/com/stories/service/StoryServiceImpl.java:16]: Please rename this instance to storiesRepository.

File: src/main/java/com/stories/service/StoryServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.stereotype.Service;
+
+import com.stories.model.StoryModel;
+import com.stories.repository.StoriesRepository;
+
+import ma.glasnost.orika.MapperFacade;
+
+@Service
+public class StoryServiceImpl implements StoryService {
+	private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(StoryServiceImpl.class);
```
Comment (soydianapinto) [inline src/main/java/com/stories/service/StoryServiceImpl.java:13]: Why are you using this logger?

File: src/main/java/com/stories/service/StoryServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+	
+	@Autowired
+	StoriesRepository repository;
+
+	@Autowired
+	private MapperFacade orikaMapperFacade;
```
Comment (soydianapinto) [inline src/main/java/com/stories/service/StoryServiceImpl.java:19]: Please rename this instance to mapperFacade

File: src/main/java/com/stories/service/StoryServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+	@Override
+	public void deleteStory(String id) {
+		if (repository.existsById(id)) {
+			log.debug("Deleting user with id: " + id);
+			repository.deleteById(id);
+		} else
+			log.error("No user was found for the given id.");
+	}
+
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/service/StoryServiceImpl.java:39]: Please delete this extra space.

File: src/main/resources/application.properties [hunk 17]
```diff
@@ -1,2 +1,2 @@
-server.port=5000
-spring.data.mongodb.uri=mongodb+srv://<username>:<password>@cluster0-nomfh.mongodb.net/InternHome
+server.port=5000
+spring.data.mongodb.uri=mongodb+srv://db_user:vloUr7AakeTEdXDj@cluster0-nomfh.mongodb.net/InternHome
```
Comment (soydianapinto) [inline src/main/resources/application.properties:2]: Please don't show the credentials.

**Review by soydianapinto — CHANGES_REQUESTED** [review]

File: src/main/java/com/stories/controller/AppController.java (hunk 6 above)
Comment (Armandosz117) [reply src/main/java/com/stories/controller/AppController.java:32]: The HTTP 200 OK success status response code indicates that the request has succeeded or 202 if the response includes an entity.

Comment (soydianapinto) [reply src/main/java/com/stories/controller/AppController.java:32]: And why are you using an ACCEPTED response?

File: src/main/java/com/stories/controller/AppController.java (hunk 4 above)
Comment (Armandosz117) [reply src/main/java/com/stories/controller/AppController.java:20]: for debugging purposes

File: src/main/java/com/stories/service/StoryServiceImpl.java (hunk 14 above)
Comment (Armandosz117) [reply src/main/java/com/stories/service/StoryServiceImpl.java:13]: for debugging purposes

File: src/main/java/com/stories/mapper/StoriesMapping.java (hunk 8 above)
Comment (Armandosz117) [reply src/main/java/com/stories/mapper/StoriesMapping.java:13]: I don't use it

**Pull request closed without merging**